graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -62,61 +88,12 @@ def merge_schemas(schema_id_to_ast):
     directives = {}  # Dict[str, DirectiveDefinition]
 
     for current_schema_id, current_ast in six.iteritems(schema_id_to_ast):
-        # Check input schema identifier is a string of alphanumeric characters and underscores
-        check_schema_identifier_is_valid(current_schema_id)
-        # Check input schema satisfies various structural requirements
-        check_ast_schema_is_valid(current_ast)
-
-        current_schema = build_ast_schema(current_ast)
-        current_query_type = get_query_type_name(current_schema)
-
-        # Merge current_ast into merged_schema_ast.
-        # Concatenate new scalars, new directives, and type definitions other than the query
-        # type to definitions list.
-        # Raise errors for conflicting scalars, directives, or types.
-        new_definitions = current_ast.definitions  # List[Node]
-        new_query_type_fields = None  # List[FieldDefinition]
-
-        for new_definition in new_definitions:
-            if isinstance(new_definition, ast_types.SchemaDefinition):
-                continue
-            elif (
-                isinstance(new_definition, ast_types.ObjectTypeDefinition) and
-                new_definition.name.value == current_query_type
-            ):  # query type definition
-                new_query_type_fields = new_definition.fields  # List[FieldDefinition]
-            elif isinstance(new_definition, ast_types.DirectiveDefinition):
-                _process_directive_definition(
-                    new_definition, directives, merged_schema_ast
-                )
-            elif isinstance(new_definition, ast_types.ScalarTypeDefinition):
-                _process_scalar_definition(
-                    new_definition, scalars, name_to_schema_id, merged_schema_ast
-                )
-            elif isinstance(new_definition, (
-                ast_types.EnumTypeDefinition,
-                ast_types.InterfaceTypeDefinition,
-                ast_types.ObjectTypeDefinition,
-                ast_types.UnionTypeDefinition,
-            )):
-                _process_generic_type_definition(
-                    new_definition, current_schema_id, scalars, name_to_schema_id, merged_schema_ast
-                )
-            else:  # All definition types should've been covered
-                raise AssertionError(
-                    u'Missed definition type: "{}"'.format(type(new_definition).__name__)
-                )
+        current_ast = deepcopy(current_ast)
+        _merge_single_schema(merged_schema_ast, name_to_schema_id, scalars, directives,
```

Comment:
nit: let's change name_to_schema_id to type_name_to_schema_id here and in other relevant places.

Reply:
What do you think of renaming _merge_single_schema to _accumulate_types ?